torch/csrc/jit/passes/liveness.cpp

```diff
@@ -88,6 +97,9 @@ struct LivenessAnalyzer {
         loop_block = processBlock(it->blocks()[0], loop_block);
         // loop block's inputs die outside loop's block
         loop_block -= toSparseBitVector(it->blocks()[0]->inputs());
+        // everything that is alive at the beginning of the loop
+        // needs to be extended until the end of the loop
+        extendLiveness(it->blocks()[0], loop_block);
```

Comment:
Liveness analysis is a classic dataflow analysis, and they all require iterating until convergence on a loop. I don't see such a convergence iteration here so I believe this is probably still incorrect and only pushes the bug down to a more complicated example.

Reply:
@zdevito added a comment